code/datums/climate.dm

```diff
@@ -20,28 +20,43 @@ var/list/weathertracker = list() //associative list, gathers time spent one each
 	var/datum/weather/current_weather
 	var/list/datum/weather/forecasts = list()
 	var/cycle_freq = list(3 MINUTES,6 MINUTES) //shortest possible time, longest possible time until next weather
+	var/z //z-level the climate is occupying
+	var/list/allowed_weather_types = list() // List of weather types this climate can have
+	var/list/weather_transitions = list() // Associative list: weather_type = list(possible_transitions)
+	var/list/weather_intensities = list() // Associative list: weather_type = intensity_level
+	var/starting_weather_type = null // The initial weather type for this climate
 
-/datum/climate/New()
+/datum/climate/New(var/active_z)
 	..()
-	if(current_weather)
+	z = active_z
```

Comment:
```suggestion
	if(active_z)
		z = active_z
	else
		z = map.zMainStation
```
or similar. I'm not sure if z is a necessary variable to the climate datum... destroy this should you find that climate datums don't need a z level (they might not, actually)